adapters/consors-bank-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ConsorsAccountInformationService.java

```diff
@@ -0,0 +1,40 @@
+package de.adorsys.xs2a.adapter.service;
+
+import de.adorsys.xs2a.adapter.adapter.BaseAccountInformationService;
+import de.adorsys.xs2a.adapter.http.HttpClient;
+import de.adorsys.xs2a.adapter.service.model.Aspsp;
+import de.adorsys.xs2a.adapter.service.model.ConsentCreationResponse;
+import de.adorsys.xs2a.adapter.service.model.Consents;
+
+import java.util.Map;
+import java.util.function.Function;
+
+public class ConsorsAccountInformationService extends BaseAccountInformationService {
+
+    public ConsorsAccountInformationService(Aspsp aspsp, HttpClient httpClient) {
+        super(aspsp, httpClient);
+    }
+
+    @Override
+    protected <T> Response<ConsentCreationResponse> createConsent(RequestHeaders requestHeaders, Consents body, Function<T, ConsentCreationResponse> mapper, HttpClient.ResponseHandler<T> responseHandler) {
+        Map<String, String> headersMap = populatePostHeaders(requestHeaders.toMap());
+        headersMap = managePsuIdHeader(headersMap);
+
+        String bodyString = jsonMapper.writeValueAsString(jsonMapper.convertValue(body, Consents.class));
+
+        Response<T> response = httpClient.post(getConsentBaseUri())
+            .jsonBody(bodyString)
+            .headers(headersMap)
+            .send(requestBuilderInterceptor, responseHandler);
+        ConsentCreationResponse creationResponse = mapper.apply(response.getBody());
+        return new Response<>(response.getStatusCode(), creationResponse, response.getHeaders());
+    }
+
+    private Map<String, String> managePsuIdHeader(Map<String, String> headers) {
```

Comment:
It seems to me that for this very case overriding the `addPsuIdHeader()` method of `AbstractService` is a good to go, so all there will be no need for overriding the entire `createConsent()` method. What do you think?

Reply:
I was thinking about that.
I agree that it would be a more elegant solution, however, the method name would not match the method behavior it that case, e.g. if there is no PSU-ID header the method will not add it.

If it sounds OK, I will gladly stick to your advice. 
Appreciate any other thoughts on this topic @spetrychenko @artemsnisarenko

Reply:
I think such behavior better apply to all adapters, thoughts?

Reply:
like adding`managePsuIdHeader()` as a new method into `AbstractService`?

Reply:
or replacing `addPsuIdHeader()`?